core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -215,8 +232,26 @@ public static String encodeNamespace(Namespace ns) {
    * @return a namespace
    */
   public static Namespace decodeNamespace(String encodedNs) {
+    return decodeNamespace(encodedNs, NAMESPACE_ESCAPED_SEPARATOR);
+  }
+
+  /**
+   * Takes in a string representation of a namespace as used for a URL parameter and returns the
+   * corresponding namespace.
+   *
+   * <p>See also {@link #encodeNamespace} for generating correctly formatted URLs.
+   *
+   * @param encodedNs a namespace to decode
+   * @param namespaceSeparator The namespace separator to use for decoding. This should be the same
+   *     separator that was used when calling {@link RESTUtil#encodeNamespace(Namespace, String)}
+   * @return a namespace
+   */
+  public static Namespace decodeNamespace(String encodedNs, String namespaceSeparator) {
```

Comment:
Nit: (I know it was like this before) Could we rename the paramaters so it's either consistently using the abbreviation or consistently using the full name rather than one parameter using one or the other? Think I'd prefer the full name `encodedNamespace` (and I don't think revAPI will complain about renaming a parameter in the existing API?)

Reply:
done